pkg/deploy/api/v1/conversion.go

```diff
@@ -143,23 +22,9 @@ func convert_v1_DeploymentTriggerImageChangeParams_To_api_DeploymentTriggerImage
 	if err := s.Convert(&in.From, &out.From, 0); err != nil {
 		return err
 	}
-	switch in.From.Kind {
-	case "DockerImage":
-		ref, err := imageapi.ParseDockerImageReference(in.From.Name)
-		if err != nil {
-			return err
-		}
-		out.Tag = ref.Tag
-		ref.Tag, ref.ID = "", ""
-		out.RepositoryName = ref.String()
-	case "ImageStreamTag":
-		name, tag, ok := imageapi.SplitImageStreamTag(in.From.Name)
-		if !ok {
-			return fmt.Errorf("ImageStreamTag object references must be in the form <name>:<tag>: %s", in.From.Name)
-		}
-		out.From.Kind = "ImageStream"
-		out.From.Name = name
-		out.Tag = tag
+	if out.From.Kind != "ImageStreamTag" {
```

Comment:
I would rather see this constructed with a positive check, like the switch/case from before.  It made it clear what our intent was and it won't accidentally make something that should be invalid, valid.